src/editor/mod.rs

```diff
@@ -282,7 +290,13 @@ impl<'arena, Node: Ast<'arena> + 'arena> Editor<'arena, Node> {
         // Show the cursor before closing so that the cursor isn't permanently disabled
         // (see issue `lotabout/tuikit#28`: https://github.com/lotabout/tuikit/issues/28)
         log::trace!("Making the cursor reappear.");
-        self.term.show_cursor(true).unwrap();
-        self.term.present().unwrap();
+        self.term.as_mut().unwrap().show_cursor().unwrap();
+        crossterm::execute!(
+            self.term.as_mut().unwrap().backend_mut(),
+            terminal::LeaveAlternateScreen
+        )
+        .unwrap();
+        self.term.as_mut().unwrap().flush().unwrap();
+        terminal::disable_raw_mode().unwrap();
```

Comment:
:laughing: This is a truly ungodly amount of unwrapping but I guess if the terminal bugs out somehow then Sapling is screwed regardless and it doesn't matter.